src/julia.h

```diff
@@ -179,6 +179,20 @@ STATIC_INLINE int jl_array_ndimwords(uint32_t ndims)
     return (ndims < 3 ? 0 : ndims-2);
 }
 
+typedef struct {
+    union {
```

Comment:
This type only exists to make it more convenient to write C functions that interact with the *data* part of the `ByteVec` type. Garbage collection is handled here:

https://github.com/JuliaLang/julia/blob/ed760e40e972fc1c6be/src/gc.c#L1631-L1636

GC needs to know whether or not to follow the pointer, which depends on the value.